### PR TITLE
add ZhipuAI integration

### PR DESCRIPTION
**Pull request by xermaor**
ZhipuAI is a large model focusing on Chinese cognition

## Summary by CodeRabbit

- **New Features**
	- Introduced serialization and deserialization for assistant messages.
	- Enhanced utility methods for data conversion in Zhipu AI processing.
	- Implemented JSON serialization/deserialization support.
	- Defined an interface for Zhipu AI service interactions.
	- Introduced classes for handling chat completions and embedding requests with Zhipu AI.
	- Provided structure for chat messages, choices, models, requests, and responses.
	- Added classes for function calls, parameters, tool interactions, and web searches within chats.
	- Established data structures for embedding information and requests.
	- Implemented builders for chat and embedding model instances.
- **Tests**
	- Added integration tests for chat model, embedding model, and streaming chat model functionalities.

### CODE REVIEW

**Comment by xermaor** [comment]
ZhipuAi Official website [https://open.bigmodel.cn/](url)

**Comment by xermaor** [comment]
@langchain4j Hi, Please review it when you have time.

**Comment by langchain4j** [comment]
Hi @1402564807, thanks a lot for your contribution! Due to a lack of time and a lot of code in your PR I will be able to review this only in 2 weeks. Thanks for understanding!

File: langchain4j-zhipu-ai/pom.xml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+        <dependency>
+            <groupId>io.jsonwebtoken</groupId>
+            <artifactId>jjwt</artifactId>
+            <version>0.12.3</version>
+        </dependency>
+
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
```
Comment (langchain4j) [inline langchain4j-zhipu-ai/pom.xml:55]: It would be great to reduce a number of dependencies. Is it even used?

**Review by ghost — APPROVED** [review]
@1402564807 thank you, great job! I count on you maintaining this module 😉

File: langchain4j-zhipu-ai/pom.xml (hunk 1 above)
Comment (langchain4j) [reply langchain4j-zhipu-ai/pom.xml:55]: Ok it was naive of me to assume that classes from guava dependency will contain "guava" in their imports

**Comment by xermaor** [comment]
> @1402564807 thank you, great job! I count on you maintaining this module 😉

Thanks, I will try my best

**Pull request merged**